.github/workflows/advanced.yml

```
@@ -14,7 +14,7 @@ jobs:
           echo "dolor sit amet" > lorem_2.md
           mkdir output  # create output dir
           # this will also include README.md
-          echo "::set-output name=files::$(printf '"%s" ' *.md)"
+          echo "files=$(printf '"%s" ' *.md)" >> $GITHUB_OUTPUT
```

```suggestion
          echo "files=$(printf '"%s" ' *.md)" > $GITHUB_OUTPUT
```

I'm not sure what you were going for here quoting wise, but I'm pretty sure this isn't what you want. How is this file parsed? Does it follow shell parsing rules (e.g. is this a POSIX variable being set?) or something else?

---

Honestly, I'm not completely sure (sorry! - this is my first open source contribution, please let me know if I have done something wrong).

I ran the action in one of my own repos and GitHub warned me that `set-output` will be depreciated soon. It should work the same as it does at the moment (just makes sure it won't stop working come June), the change is as suggested in the actions documentation - see [here](https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/#examples) and [here](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-output-parameter).

---

Yes I do understand this needs to be updated to stop throwing warnings and not stop working in the near future. The issue her is what that combination of `echo` and `prinf` are going to do. This will output something like this:

```
file="file1.md" "file2.md" "file3.md" 
```

The issue here is that the variable value probably needs to be a single string like this:

```
file="file1.md file2.md file3.md"
```

To get that we'd probably want code more like this:

```sh
echo 'files="'$(echo *.md)'"' > $GITHUB_OUTPUT
```

But since I'm not 100% sure on how this file is being parsed I can's say for sure if that's correct or how we would need to account for spaces in filenames (i.e. what escaping style we should use for the filenames).

---

Thank you for the clarification! 😃 
I have just tested the action with both outputs.

--------
When using the current code:
```bash
echo "files=$(printf '"%s" ' *.md)" >> $GITHUB_OUTPUT
```
or
```bash
echo "files=$(printf '"%s" ' *.md)" > $GITHUB_OUTPUT
```
The value of `steps.files_list.outputs.files` is as you expected:
```
"file1.md" "file2.md" "file3.md"
```
The action is successful and the PDF file builds as expected

--------
However, when running the action with the code below, so all files are in a single string:
```bash
echo 'files="'$(echo *.md)'"' > $GITHUB_OUTPUT
```
The action fails with the error:
```
pandoc: file1.md file2.md file3.md ..etc.. file50.md: openBinaryFile: invalid argument (File name too long)
```

---

Fair enough. Weird but fair. That suggests they are *not* using standard shell quoting for this, so I really don't know what the file parsing is like, but as long as your expected result of all three inputs getting concatenated is working we can go ahead with it like you had it (with the small change of using `>` so we don't append the file but just write it; In this case it shouldn't matter but if mixed and matched with other actions it might matter that we don't overwrite the file).

---

Cool, thank you, and apologies for my confusion earlier! 😄